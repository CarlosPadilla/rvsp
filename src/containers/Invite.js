import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as actions from '../actions/invite';
import { maxNumber } from '../helpers';
import InviteList from '../components/InviteList/InviteList';

class Invite extends Component {
  state = {
    name: '',
    email: '',
    invites: [],
  }

  onInputChange = (event) => {
    this.setState({
      [event.target.id]: event.target.value,
    });
  }

  onSendHandler = (event) => {
    event.preventDefault();
    const { invites } = this.state;
    const send = {
      id: maxNumber(this.state.invites.map(invite => invite.id)) + 1,
      name: this.state.name,
      email: this.state.email,
    };
    invites.push(send);
    this.props.onSend({ email: this.state.email, name: this.state.name });
    this.setState({ name: '', email: '' });
  }

  render() {
    return (
      <div>
        <h1>Invite</h1>
        <div>
          <label>Nimi</label>
          <input id="name" className="input-name" type="text" value={this.state.name} onChange={this.onInputChange} />
        </div>
        <div>
          <label>Email</label>
          <input id="email" className="input-email" type="email" value={this.state.email} onChange={this.onInputChange} />
        </div>
        <button onClick={this.onSendHandler}>Saada kutse!</button>
        <div>
          <InviteList invites={this.state.invites} />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  invites: state.invites,
});

const mapDispatchToProps = dispatch => ({
  onSend: invite => dispatch(actions.sendInvite(invite)),
});

Invite.propTypes = {
  onSend: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Invite);
