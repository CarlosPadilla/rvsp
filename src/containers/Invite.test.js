import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Invite from './Invite';

const store = configureStore([
    thunk,
])();

describe('Invite', () => {

  const invite = shallow(<Invite store={store} />).dive()

  it('renders', () => {
    expect(invite).toMatchSnapshot()
  })

  it('initializes the state with empty string of name and email', () => {
    expect(invite.state().name).toEqual('')
    expect(invite.state().email).toEqual('')
  })

  describe('when typing in to name input', () => {

    const name = 'Aimar'
    const id = 'name'

    beforeEach(() => {
      invite.find('.input-name').simulate('change', { target: { value: name, id: id }})
    })

    it('updates the name in `state`', () => {
      expect(invite.state().name).toEqual(name)
    })
  })

  describe('when typing in to email input', () => {

    const email = 'aimar@snapit.ee'
    const id = 'email'

    beforeEach(() => {
      invite.find('.input-email').simulate('change', { target: { value: email, id: id }})
    })

    it('updates the email in `state`', () => {
      expect(invite.state().email).toEqual(email)
    })

  })

})
