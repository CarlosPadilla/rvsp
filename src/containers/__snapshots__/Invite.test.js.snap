// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invite renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Invite
    invites={undefined}
    onSend={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": null,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Invite
</h1>,
        <div>
          <label>
                    Nimi
          </label>
          <input
                    className="input-name"
                    id="name"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
</div>,
        <div>
          <label>
                    Email
          </label>
          <input
                    className="input-email"
                    id="email"
                    onChange={[Function]}
                    type="email"
                    value=""
          />
</div>,
        <button
          onClick={[Function]}
>
          Saada kutse!
</button>,
        <div>
          <InviteList
                    invites={Array []}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Invite",
        },
        "ref": null,
        "rendered": "Invite",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Nimi
</label>,
            <input
              className="input-name"
              id="name"
              onChange={[Function]}
              type="text"
              value=""
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Nimi",
            },
            "ref": null,
            "rendered": "Nimi",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-name",
              "id": "name",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Email
</label>,
            <input
              className="input-email"
              id="email"
              onChange={[Function]}
              type="email"
              value=""
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
            },
            "ref": null,
            "rendered": "Email",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-email",
              "id": "email",
              "onChange": [Function],
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Saada kutse!",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Saada kutse!",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <InviteList
            invites={Array []}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "invites": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Invite
</h1>,
          <div>
            <label>
                        Nimi
            </label>
            <input
                        className="input-name"
                        id="name"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
</div>,
          <div>
            <label>
                        Email
            </label>
            <input
                        className="input-email"
                        id="email"
                        onChange={[Function]}
                        type="email"
                        value=""
            />
</div>,
          <button
            onClick={[Function]}
>
            Saada kutse!
</button>,
          <div>
            <InviteList
                        invites={Array []}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Invite",
          },
          "ref": null,
          "rendered": "Invite",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Nimi
</label>,
              <input
                className="input-name"
                id="name"
                onChange={[Function]}
                type="text"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Nimi",
              },
              "ref": null,
              "rendered": "Nimi",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-name",
                "id": "name",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Email
</label>,
              <input
                className="input-email"
                id="email"
                onChange={[Function]}
                type="email"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-email",
                "id": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Saada kutse!",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Saada kutse!",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <InviteList
              invites={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "invites": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
